Still this compile error on windows:

```
ninoLocal@NINO-WIN10 MINGW64 ~/codeProjects/wheelsprung (main)
$ nimble clean
  Executing task clean in C:\Users\ninov\codeProjects\wheelsprung\wheelsprung.nimble

ninoLocal@NINO-WIN10 MINGW64 ~/codeProjects/wheelsprung (main)
$ nimble install
  Verifying dependencies for wheelsprung@0.2.0
      Info: Dependency on https://github.com/samdze/playdate-nim@#finish-callback-fix already satisfied
  Verifying dependencies for playdate@#finish-callback-fix
      Info: Dependency on https://github.com/ninovanhooff/nim-chipmunk-playdate@any version already satisfied
  Verifying dependencies for chipmunk7@7.0.3
 Installing wheelsprung@0.2.0
   Building wheelsprung/wheelsprung.exe using c backend
C:\Users\ninov\nimcache\wheelsprung_r\@m..@s..@s..@s.nimble@spkgs@splaydate-@hfinish-callback-fix@splaydate@ssound.nim.c: In function 'setFinishCallback__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZsound_569':
C:\Users\ninov\nimcache\wheelsprung_r\@m..@s..@s..@s.nimble@spkgs@splaydate-@hfinish-callback-fix@splaydate@ssound.nim.c:759:18: error: too many arguments to function '((const struct playdate_sound_sampleplayer *)playdate__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZbindingsZapi_30->sound->sampleplayer)->setFinishCallback'
  759 |                 (*(*(*playdate__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZbindingsZapi_30).sound).sampleplayer).setFinishCallback((*this_0).Sup.resource, NIM_NIL, NIM_NIL);
      |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:\Users\ninov\nimcache\wheelsprung_r\@m..@s..@s..@s.nimble@spkgs@splaydate-@hfinish-callback-fix@splaydate@ssound.nim.c:765:18: error: too many arguments to function '((const struct playdate_sound_sampleplayer *)playdate__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZbindingsZapi_30->sound->sampleplayer)->setFinishCallback'
  765 |                 (*(*(*playdate__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZbindingsZapi_30).sound).sampleplayer).setFinishCallback((*this_0).Sup.resource, privateSamplePlayerFinishCallback__OOZOOZOOZOnimbleZpkgsZplaydate4535finish45callback45fixZplaydateZsound_559, ((void*) (this_0)));
      |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Error: execution of an external compiler program 'C:\nim-1.6.16\dist\mingw64\bin\gcc.exe -c  -w -fmax-errors=3 -mno-ms-bitfields -DTARGET_EXTENSION=1 -Wall -Wno-unknown-pragmas -Wdouble-promotion -IC:\Users\ninov\Documents\PlaydateSDK\C_API -O3 -fno-strict-aliasing -fno-ident   -IC:\nim-1.6.16\lib -IC:\Users\ninov\codeProjects\wheelsprung\src -o C:\Users\ninov\nimcache\wheelsprung_r\@m..@s..@s..@s.nimble@spkgs@splaydate-@hfinish-callback-fix@splaydate@ssound.nim.c.o C:\Users\ninov\nimcache\wheelsprung_r\@m..@s..@s..@s.nimble@spkgs@splaydate-@hfinish-callback-fix@splaydate@ssound.nim.c' failed with exit code: 1


       Tip: 5 messages have been suppressed, use --verbose to show them.
     Error: Build failed for package: wheelsprung
        ... Execution failed with exit code 1
        ... Command: C:\nim-1.6.16\bin\nim.exe c --colors:on --noNimblePath -d:release -d:NimblePkgVersion=0.2.0 --path:C:\Users\ninov\.nimble\pkgs\playdate-#finish-callback-fix --path:C:\Users\ninov\.nimble\pkgs\chipmunk7-7.0.3 --hints:off -o:C:\Users\ninov\codeProjects\wheelsprung\wheelsprung.exe C:\Users\ninov\codeProjects\wheelsprung\src\wheelsprung.nim

ninoLocal@NINO-WIN10 MINGW64 ~/codeProjects/wheelsprung (main)
$
```

--- a/playdate_example/src/playdate_example.nim
+++ b/playdate_example/src/playdate_example.nim
@@ -74,6 +74,9 @@ proc handler(event: PDSystemEvent, keycode: uint) {.raises: [].} =
         # Errors are handled through exceptions
         try:
             samplePlayer = playdate.sound.newSamplePlayer("/audio/jingle")
+
+            samplePlayer.finishCallback = proc(player: SamplePlayer) =
+                playdate.system.logToConsole("Sound finished playing.")
         except:
             playdate.system.logToConsole(getCurrentExceptionMsg())
         # Inline try/except

--- a/src/playdate/bindings/sound.nim
+++ b/src/playdate/bindings/sound.nim
@@ -5,9 +5,10 @@ import utils
 type FilePlayerPtr = pointer
 type AudioSamplePtr = pointer
 type SamplePlayerPtr = pointer
-type SoundSourcePtr = pointer
+type SoundSourceRaw {.importc: "SoundSource", header: "pd_api.h".} = object
+type SoundSourcePtr = ptr SoundSourceRaw
 
-type PDSndCallbackProcRaw {.importc: "sndCallbackProc", header: "pd_api.h".} = proc(soundSource: SoundSourcePtr, userData: pointer): void {.cdecl.}
+type PDSndCallbackProcRaw {.importc: "sndCallbackProc", header: "pd_api.h".} = proc(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl.}
 
 
 type PlaydateSoundFileplayer {.importc: "const struct playdate_sound_fileplayer",
@@ -33,8 +34,8 @@ type PlaydateSoundFileplayer {.importc: "const struct playdate_sound_fileplayer"
     # setLoopRange* {.importc: "setLoopRange".}: proc (player: ptr FilePlayer;
     #     start: cfloat; `end`: cfloat) {.cdecl.}
     # didUnderrun* {.importc: "didUnderrun".}: proc (player: ptr FilePlayer): cint {.cdecl.}
-    # setFinishCallback* {.importc: "setFinishCallback".}: proc (
-    #     player: ptr FilePlayer; callback: SndCallbackProc) {.cdecl.}
+    setFinishCallback* {.importc: "setFinishCallback".}: proc (
+        player: FilePlayerPtr; callback: PDSndCallbackProcRaw, userData: pointer = nil) {.cdecl, raises: [].}
     # setLoopCallback* {.importc: "setLoopCallback".}: proc (player: ptr FilePlayer;
     #     callback: SndCallbackProc) {.cdecl.}
     getOffset {.importc: "getOffset".}: proc (player: FilePlayerPtr): cfloat {.cdecl, raises: [].}

--- a/src/playdate/sound.nim
+++ b/src/playdate/sound.nim
@@ -49,13 +49,15 @@ proc getLength*(this: AudioSample): float32 =
 # AudioSource
 type SoundSourceObj {.requiresinit.} = object of RootObj
     resource: pointer
-    callback: PDSoundCallbackFunction
 type SoundSource* = ref SoundSourceObj
 
 # FilePlayer
 type
     FilePlayerObj = object of SoundSourceObj
+        finishCallback: PDFilePlayerCallbackFunction
     FilePlayer* = ref FilePlayerObj
+
+    PDFilePlayerCallbackFunction* = proc(player: FilePlayer) {.raises: [].}
 
 proc `=destroy`(this: var FilePlayerObj) =
     privateAccess(PlaydateSound)
@@ -65,12 +67,12 @@ proc `=destroy`(this: var FilePlayerObj) =
 proc newFilePlayer*(this: ptr PlaydateSound): FilePlayer =
     privateAccess(PlaydateSound)
     privateAccess(PlaydateSoundFileplayer)
-    result = FilePlayer(resource: this.fileplayer.newPlayer(), callback: nil)
+    result = FilePlayer(resource: this.fileplayer.newPlayer(), finishCallback: nil)
 
 proc newFilePlayer*(this: ptr PlaydateSound, path: string): FilePlayer {.raises: [IOError].} =
     privateAccess(PlaydateSound)
     privateAccess(PlaydateSoundFileplayer)
-    result = FilePlayer(resource: this.fileplayer.newPlayer(), callback: nil)
+    result = FilePlayer(resource: this.fileplayer.newPlayer(), finishCallback: nil)
     if this.fileplayer.loadIntoPlayer(result.resource, path.cstring) == 0:
         raise newException(IOError, fmt"file {path} not found: No such file")
 
@@ -132,11 +134,34 @@ proc `offset=`*(this: FilePlayer, offset: float32) =
     privateAccess(PlaydateSoundFileplayer)
     playdate.sound.fileplayer.setOffset(this.resource, offset.cfloat)
 
+proc privateFilePlayerFinishCallback(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl, raises: [].} =
+    let filePlayer = cast[FilePlayer](userdata)
+    if filePlayer.finishCallback != nil:
+        filePlayer.finishCallback(filePlayer)
+
+proc setFinishCallback*(this: FilePlayer, callback: PDFilePlayerCallbackFunction) =
+    privateAccess(PlaydateSound)
+    privateAccess(PlaydateSoundFileplayer)
+    this.finishCallback = callback
+    if callback == nil:
+        playdate.sound.fileplayer.setFinishCallback(this.resource, nil, nil)
+    else:
+        playdate.sound.fileplayer.setFinishCallback(this.resource, privateFilePlayerFinishCallback, cast[pointer](this))
+
+proc finishCallback*(this: FilePlayer): PDFilePlayerCallbackFunction =
+    return this.finishCallback
+
+proc `finishCallback=`*(this: FilePlayer, callback: PDFilePlayerCallbackFunction) =
+    this.setFinishCallback(callback)
+
 # SamplePlayer
 type
     SamplePlayerObj = object of SoundSourceObj
         sample: AudioSample
+        finishCallback: PDSamplePlayerCallbackFunction
     SamplePlayer* = ref SamplePlayerObj
+
+    PDSamplePlayerCallbackFunction* = proc(player: SamplePlayer) {.raises: [].}
 
 proc `=destroy`(this: var SamplePlayerObj) =
     privateAccess(PlaydateSound)
@@ -147,7 +172,7 @@ proc `=destroy`(this: var SamplePlayerObj) =
 proc newSamplePlayer*(this: ptr PlaydateSound): SamplePlayer =
     privateAccess(PlaydateSound)
     privateAccess(PlaydateSoundSampleplayer)
-    result = SamplePlayer(resource: this.sampleplayer.newPlayer(), callback: nil)
+    result = SamplePlayer(resource: this.sampleplayer.newPlayer(), finishCallback: nil)
 
 proc sample*(this: SamplePlayer): AudioSample =
     return this.sample
@@ -161,7 +186,7 @@ proc `sample=`*(this: SamplePlayer, sample: AudioSample) =
 proc newSamplePlayer*(this: ptr PlaydateSound, path: string): SamplePlayer {.raises: [IOError].} =
     privateAccess(PlaydateSound)
     privateAccess(PlaydateSoundSampleplayer)
-    result = SamplePlayer(resource: this.sampleplayer.newPlayer(), callback: nil)
+    result = SamplePlayer(resource: this.sampleplayer.newPlayer(), finishCallback: nil)
     result.`sample=`(this.newAudioSample(path))
 
 proc volume*(this: SamplePlayer): tuple[left: float32, right: float32] =
@@ -221,25 +246,25 @@ proc rate*(this: SamplePlayer): float32 =
     privateAccess(PlaydateSoundSampleplayer)
     return playdate.sound.sampleplayer.getRate(this.resource).float32
 
-proc privateSampleFinishCallback(soundSourcePtr: SoundSourcePtr, userData: pointer) {.cdecl, raises: [].} =
-    try:
-        let samplePlayer = cast[SamplePlayer](userdata)[]
-        if samplePlayer.callback != nil:
-            samplePlayer.callback(userData)
-    except:
-        echo "Error while calling SamplePlayer finish callback"
+proc privateSamplePlayerFinishCallback(soundSource: SoundSourcePtr, userdata: pointer) {.cdecl, raises: [].} =
+    let samplePlayer = cast[SamplePlayer](userdata)
+    if samplePlayer.finishCallback != nil:
+        samplePlayer.finishCallback(samplePlayer)
 
-proc setFinishCallback*(this: SamplePlayer, callback: PDSoundCallbackFunction) =
+proc setFinishCallback*(this: SamplePlayer, callback: PDSamplePlayerCallbackFunction) =
     privateAccess(PlaydateSound)
     privateAccess(PlaydateSoundSampleplayer)
-    try:
-        this.callback = callback
-        if callback == nil:
-            playdate.sound.sampleplayer.setFinishCallback(this.resource, nil, nil)
-        else:
-            playdate.sound.sampleplayer.setFinishCallback(this.resource, privateSampleFinishCallback, cast[pointer](this))
-    except:
-        echo "Error setting finish callback"
+    this.finishCallback = callback
+    if callback == nil:
+        playdate.sound.sampleplayer.setFinishCallback(this.resource, nil, nil)
+    else:
+        playdate.sound.sampleplayer.setFinishCallback(this.resource, privateSamplePlayerFinishCallback, cast[pointer](this))
+
+proc finishCallback*(this: SamplePlayer): PDSamplePlayerCallbackFunction =
+    return this.finishCallback
+
+proc `finishCallback=`*(this: SamplePlayer, callback: PDSamplePlayerCallbackFunction) =
+    this.setFinishCallback(callback)
 
 proc setPlayRange*(this: SamplePlayer, start: int32, `end`: int32) =
     privateAccess(PlaydateSound)

